### DISCUSSION TITLE
Is there a way to prevent text from wrapping?

### DISCUSSION THREAD

**Discussion by hyunrealshadow** [typst/typst#2915] [Q&A] [answered]
I now use the scale function to achieve the font scaling effect
Just like the effect in the picture below in Word
![image](https://github.com/typst/typst/assets/36305150/873c00ce-4d02-4fbe-b9ed-ee0d51ba34e8)
But if you use the scale function, it will wrap first and then scale
Just like the effect in the picture below
![image](https://github.com/typst/typst/assets/36305150/82d0303b-d8e0-4791-bf1b-f40becac9fb0)
Because my font does not support stretch
I can do this using two places.
```typst
    #block(height: 22pt, width: 100%, spacing: 0pt)[
      #set text(font: "Source Han Sans", size: 21pt, weight: "bold")
      #set align(start + horizon)
      #set par(justify: true, linebreaks: "optimized")
      #place(top + left, clearance: 0pt, dx: 0pt, dy: 2pt)[#box(width: 100%)[
        #scale(x: 58%, y: 100%, origin: left + horizon)[一行很长的字体，我希望它不要换行显示，]
      ]]
      #place(top + right, clearance: 0pt, dx: 0pt, dy: 2pt)[#box(width: 100%)[
        #scale(x: 58%, y: 100%, origin: right + horizon)[就像Word中一样，我希望实现压缩字体效果]
      ]]
    ]
```
As shown below
![image](https://github.com/typst/typst/assets/36305150/81562be1-a026-4797-872f-4efe6a3fed12)
But in the current version (0.10.0), is there any other way?

**Comment by Enivex** [comment, marked as answer]
```#scale(x: 58%,origin: left,block(width: 200%)[一行很长的字体，我希望它不要换行显示，就像Word中一样，我希望实现压缩字体效果])```

should work
(Set 200% to big enough that the line doesn't wrap)

**Reply by hyunrealshadow** [reply]
It works for me, Thank you
